NickvisionTubeConverter.GNOME/Views/MainWindow.cs

```diff
@@ -50,162 +35,57 @@ public MainWindow(MainWindowController controller, Adw.Application application)
-        _clampStartPage.SetMaximumSize(450);
-        _clampStartPage.SetValign(Gtk.Align.Center);
-        _clampStartPage.SetMarginStart(12);
-        _clampStartPage.SetMarginEnd(12);
-        _clampStartPage.SetMarginTop(12);
-        _clampStartPage.SetMarginBottom(12);
-        _scrollStartPage.SetChild(_clampStartPage);
-        _boxStartPage = Gtk.Box.New(Gtk.Orientation.Vertical, 12);
-        _boxStartPage.SetHexpand(true);
-        _boxStartPage.SetHalign(Gtk.Align.Fill);
-        _boxStartPage.Append(_greetingStartPage);
-        _boxStartPage.Append(_btnAddDownloadStartPage);
-        _boxStartPage.Append(_lblStart);
-        _clampStartPage.SetChild(_boxStartPage);
-        //Downloads Page
-        _scrollDownloadsPage = Gtk.ScrolledWindow.New();
-        _clampDownloadsPage = Adw.Clamp.New();
-        _clampDownloadsPage.SetMaximumSize(800);
-        _clampDownloadsPage.SetMarginStart(12);
-        _clampDownloadsPage.SetMarginEnd(12);
-        _clampDownloadsPage.SetMarginTop(12);
-        _clampDownloadsPage.SetMarginBottom(12);
-        _scrollDownloadsPage.SetChild(_clampDownloadsPage);
-        _boxMainContent = Gtk.Box.New(Gtk.Orientation.Vertical, 10);
-        _boxMainContent.SetHexpand(true);
-        _boxMainContent.SetValign(Gtk.Align.Start);
-        _clampDownloadsPage.SetChild(_boxMainContent);
-        _lblDownloads = Gtk.Label.New(_controller.Localizer["Downloads"]);
-        _lblDownloads.SetHalign(Gtk.Align.Start);
-        _lblDownloads.SetMarginStart(5);
-        _lblDownloads.AddCssClass("heading");
-        _boxMainContent.Append(_lblDownloads);
-        _boxDownloads = Gtk.Box.New(Gtk.Orientation.Vertical, 0);
-        _boxDownloads.SetHexpand(true);
-        _boxDownloads.SetValign(Gtk.Align.Start);
-        _boxDownloads.AddCssClass("card");
-        _boxMainContent.Append(_boxDownloads);
-        //View Stack
-        _viewStack = Adw.ViewStack.New();
-        _viewStack.AddNamed(_scrollStartPage, "pageNoDownloads");
-        _viewStack.AddNamed(_scrollDownloadsPage, "pageDownloads");
-        _toastOverlay.SetChild(_viewStack);
-        //Spinner Box
-        _binSpinner = Adw.Bin.New();
-        _binSpinner.SetHexpand(true);
-        _binSpinner.SetVexpand(true);
-        //Spinner
-        _spinner = Gtk.Spinner.New();
-        _spinner.SetSizeRequest(48, 48);
-        _spinner.SetHalign(Gtk.Align.Center);
-        _spinner.SetValign(Gtk.Align.Center);
-        _spinner.SetHexpand(true);
-        _spinner.SetVexpand(true);
-        _binSpinner.SetChild(_spinner);
-        //Loading Overlay
-        _overlayLoading = Gtk.Overlay.New();
-        _overlayLoading.SetVexpand(true);
-        _overlayLoading.AddOverlay(_binSpinner);
-        _overlayLoading.SetChild(_mainBox);
-        Handle.SetContent(_overlayLoading);
+        OnCloseRequest += OnCloseRequested;
+        //Build UI
+        Gtk.Builder builder = Builder.FromFile("window.ui", _controller.Localizer);
```

Comment:
is explicit type required? The method can only return `Gtk.Builder`, so `var` here should work I think